### PR TITLE
feat: added operating-system input variable

### PR DESCRIPTION
**Pull request by Khuzaima05**
### Description
Add operating-system input variable for OCP 4.15 and beyond (RHCoreOS now an option)

issue: https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/issues/429

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by Khuzaima05** [comment]
Pipeline is failing with the following error:

`Error: Request failed with status code: 400, ServerErrorResponse: {"incidentID":"524af746-bd3d-4e16-9883-9ee890cc7813,524af746-bd3d-4e16-9883-9ee890cc7813","code":"E1437","description":"The action cannot be completed because it would reduce the worker pool size to below the cluster's minimum requirement of 2 workers.","type":"BadRequest"}`

Other [PR](https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/actions/runs/9174824931/job/25226696386#step:7:4409) pipeline also facing same error.

File: variables.tf [hunk 1]
```diff
@@ -249,6 +249,11 @@ variable "disable_outbound_traffic_protection" {
   default     = false
 }
 
+variable "operating_system" {
+  type        = string
+  description = "The operating system of the workers in the default worker pool."
```
Comment (ocofaigh) [inline variables.tf:254]: can you include in the description what the behaviour is when value is not set?

File: variables.tf [hunk 2]
```diff
@@ -249,6 +249,12 @@ variable "disable_outbound_traffic_protection" {
   default     = false
 }
 
+variable "operating_system" {
+  type        = string
+  description = "The operating system of the workers in the default worker pool. If no value is specified, RHEL 8 will be used as the default."
```
Comment (ocofaigh) [inline variables.tf:254]: surely it won't always be RHEL 8. I guess it will default to whatever the current default OS is.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
This needs to be exposed in fscloud submodule too

**Review by ocofaigh — APPROVED** [review]

**Review by vburckhardt — COMMENTED** [review]
Missing the aspect about pool - https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/container_vpc_worker_pool#operating_system

File: variables.tf [hunk 3]
```diff
@@ -249,6 +249,12 @@ variable "disable_outbound_traffic_protection" {
   default     = false
 }
 
+variable "operating_system" {
+  type        = string
+  description = "The operating system of the workers in the default worker pool. If no value is specified, the current default version OS will be used. See https://cloud.ibm.com/docs/openshift?topic=openshift-openshift_versions#openshift_versions_available ."
+  default     = null
+}
```
[the remark this answers is not in the document]
Comment (imprateeksh) [reply variables.tf:256]: This will be addressed in the [PR-466](https://github.com/terraform-ibm-modules/terraform-ibm-base-ocp-vpc/pull/466)

**Pull request merged**